test/benchmark/default_simulator.py

```diff
@@ -0,0 +1,281 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019, 2020, 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+import numpy as np
+from time import time
+from abc import ABC
+
+from qiskit import QuantumCircuit, transpile
+import qiskit.quantum_info as qi
+from qiskit.circuit.library import QuantumVolume, QFT, RealAmplitudes
+from qiskit.providers.aer.noise import NoiseModel, ReadoutError, amplitude_damping_error, depolarizing_error, phase_amplitude_damping_error
+
+class _Base(ABC):
+
+    def __init__(self,
+                 qubits=[5, 15, 25]):
+
+        self._simulator = None
+        self.params = (qubits)
+        self.param_names = ["qubits"]
+        
+    def simulator(self):
+        if self._simulator:
+            return self._simulator
+        try:
+            from qiskit.providers.aer import AerSimulator
+            self._simulator = AerSimulator()
+        except:
+            from qiskit.providers.aer import QasmSimulator
+            self._simulator = QasmSimulator()
+        return self._simulator
+
+    def _run(self, circuit, *args, **kwargs):
+        if self.simulator().__class__.__name__ == 'AerSimulator':
+            return self.simulator().run(circuit, *args, **kwargs).result()
+        elif self.simulator().__class__.__name__ in ('QasmSimulator', 'UnitarySimulator'):
+            from qiskit import assemble
+            circuit = assemble(circuit, self.simulator())
+            return self.simulator().run(circuit, *args, **kwargs).result()
+        else:
+            raise ValueError(f'unknown simulator class: {self._simulator.__class__.__name__}')
+
+
+class Benchmark(_Base):
+
+    def __init__(self,
+                 qubits=[5, 15, 25]):
+        
+        super().__init__(qubits)
+
+    def _track(self, circuit):
+        """track only simulation time of given circuit"""
+
+        self.add_measure(circuit)
+
+        circuit = transpile(circuit, self.simulator())
+
+        if circuit.num_parameters > 0:
+            params = [ np.random.random() for _ in range(circuit.num_parameters) ]
+            circuit = circuit.bind_parameters(params)
+
+        # benchmark start
+        start_ts = time()
+        result = self._run(circuit)
+        end_ts = time()
+        # benchmark end
+
+        if result.success:
+            return end_ts - start_ts
+        else:
+            raise ValueError(result.status)
+
```

Comment:
This is a pretty large subversion of how `asv` operates, and performance timing with `time.time` is not reliable at all; it's subject to the system clock, and there's no guarantee that it'll have a granularity of better than a second (though in practice it usually will).  We should use proper `time_*` methods and setup functions - you could separate out each one into its own class, and inherit a `setup` function that calls a method `circuit` which you define in each class separately.

I don't know if you can tag individual class methods with `setup` functions in asv, but if you can, we may be able to do it in a single class compactly too.  Docs: https://asv.readthedocs.io/en/stable/writing_benchmarks.html#setup-and-teardown-functions